src/Routes/RoutesFilter.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mezzio\Tooling\Routes;
+
+use ArrayIterator;
+use FilterIterator;
+use Mezzio\Router\Route;
+
+use function array_intersect;
+use function array_walk;
+use function get_class;
+use function in_array;
+use function is_array;
+use function is_string;
+use function preg_match;
+use function sprintf;
+use function str_replace;
+use function strtoupper;
+
+/**
+ * RoutesFilter filters a traversable list of Route objects based on any of the four Route criteria,
+ * those being the route's name, path, middleware, or supported method(s).
+ */
+class RoutesFilter extends FilterIterator
```

Comment:
you should provide what the iterator is actually about. Usually, this is done via generics:

```php
/**
 * @template-extends FilterIterator<array-key,Route>
 */
final class RoutesFilter extends FilterIterator
{}
```

I just used `array-key` because I guess we don't mind about the key type.